lib/lightning_web/live/workflow_live/edit.ex

```diff
@@ -703,6 +707,14 @@ defmodule LightningWeb.WorkflowLive.Edit do
         {:noreply,
          socket
          |> put_flash(:error, "You can't delete the first step of a workflow.")}
+
+      :has_steps ->
+        {:noreply,
+         socket
+         |> put_flash(
+           :error,
+           "You can't delete a step that has already been ran."
+         )}
```

Comment:
I don't think we should do anything about this now, but I think we're at the point where this logic needs to be moved out into something that acts as the "source of truth" for all this behaviour, this is business logic that exists only in one place.

Reply:
Okay got it, I am noting it somewhere so that we can chat about it in one of our 1-1s.